lib/src/components/NxToast/NxToast.scss

```
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+.nx-toast__wrapper {
+  ///nx-toast__wrapper has position absolute to ensure toasts stay in the top right corner of the page.
+  //Using display:flex and flex-end ensures the right side of the toast is aligned with the right side of 
+  //nx-page-main. The nx-toast_container has position fixed applied so it remains fixed while scrolling.
+  position: absolute;
+  top: 0;
+  right: 0;
+  display: flex;
+  justify-content: flex-end;
+  margin-right: var(--nx-spacing-6x);
+}
+
+.nx-toast__container {
+  position: fixed;
+  width: 500px;
+}
+
+.nx-toast {
+  position: relative;
+  animation: animationIn 0.4s linear;
+
+  &.nx-toast--closing {
+    animation: animationOut 0.4s linear;
+  }
+  
+  .nx-alert {
+    margin: var(--nx-spacing-2x) 0 0 0;
+  }
+}
+
+@keyframes animationIn {
+  0% {
+    margin-top: 0;
+    max-height: 0px;
+    right: -500px;
+    opacity: 0;
+  }
+  40% {
+    margin-top: var(--nx-spacing-2x);
+    right: -500px;
+    opacity: 0;
+    max-height:74px;
+  }
+  100% {
+    right:0;
+    opacity: 1;
+  }
+}
+
+@keyframes animationOut {
+  0% {
+    max-height: 75px;
+    right: 0;
+    opacity: 1;
+  }
+  75% {
+    max-height:74px;
+    right: -500px;
+    opacity: 0;
+  }
+  100% {
+    max-height: 0px;
+    right:-500px;
+    opacity: 0;
+  }
+}
+
+/*
+  If the user has expressed their preference for reduced motion,
+  then only animate the fade effect.
+*/
+@media (prefers-reduced-motion: reduce) {
+  .nx-toast {
+    animation: reducedAnimationIn 0.4s linear;
+
+    &.nx-toast--closing {
+      animation: reducedAnimationOut 0.4s linear;
```

These could both use the same animation and `nx-toast--closing` could just set `animation-direction: reverse`

---

> These could both use the same animation and nx-toast--closing could just set animation-direction: reverse

Nevermind, that doesn't seem to work

---

> > These could both use the same animation and nx-toast--closing could just set animation-direction: reverse
>
> Nevermind, that doesn't seem to work

I figured out a way to do it, but the syntax is a little weird.  Still probably worth it IMO to get rid of the extra `@keyframes` declaration.  It'd be like this, though IMO we'd want to adjust the name in this case
```suggestion
    animation: reducedAnimationIn 0.4s linear;

    &.nx-toast--closing {
            animation: reducedAnimationIn, reducedAnimationIn 0.4s linear reverse;
```

---

> > > These could both use the same animation and nx-toast--closing could just set animation-direction: reverse
> > 
> > 
> > Nevermind, that doesn't seem to work
> 
> I figured out a way to do it, but the syntax is a little weird. Still probably worth it IMO to get rid of the extra `@keyframes` declaration. It'd be like this, though IMO we'd want to adjust the name in this case

When fiddling with this, your solution works cleanly on Chrome, but for some reason didn't work on Firefox and Safari on my Mac. Where did you see this syntax, and I can look into why this isn't working on other browsers and get that resolved.

---

> When fiddling with this, your solution works cleanly on Chrome, but for some reason didn't work on Firefox and Safari on my Mac. Where did you see this syntax, and I can look into why this isn't working on other browsers and get that resolved.

Oh in that case it's probably not worth worrying about.  But I worked it out after reading specs and documentation and then experimenting.  To be honest though, it seemed backwards from how the spec said it was supposed to work, so maybe Chrome is actually wrong.  If you want to experiment further, you can try the following in the other browsers (it didn't work in Chrome) which is what I _thought_ was supposed to be right based on the spec:
```
animation: reducedAnimationIn 0.4s linear reverse, reducedAnimationIn;
```

The most relevant thing I read was the last paragraph before the big example in this section of the spec: https://drafts.csswg.org/css-animations/#animations